src/leader.h

```diff
@@ -63,8 +66,12 @@ struct exec {
 	uint64_t id;
 	int status;
 	queue queue;
-	exec_cb cb;
 	pool_work_t work;
+	/* When the callback is invoked, this field is `true` if raft_barrier
+	 * or raft_apply was called and `false` if the callback was invoked
+	 * immediately. */
```

Comment:
IMO this comment is talking about how this field is set from outside not about what the field means or how it is used by `struct exec`.

Reply:
I think it makes sense as-is, the field is set by code in leader.c and it's part of the API of that translation unit as seen by the gateway. Maybe there's a way I could reword it that would make more sense?

Reply:
Rewrote this, also removed the obsolete comment on leader__exec.